src/ucp/core/ucp_mm.c

```diff
@@ -783,7 +865,7 @@ ucp_mpool_free(ucp_worker_h worker, ucs_mpool_t *mp, void *chunk)
     ucp_mem_desc_t *chunk_hdr;
 
     chunk_hdr = (ucp_mem_desc_t*)chunk - 1;
-    ucp_memh_put(worker->context, chunk_hdr->memh, 0);
+    ucp_memh_put(worker->context, chunk_hdr->memh, 1);
```

Comment:
ucp_memh_alloc should also create "indirect" memh , therefore would not need to pass invalidate=1 here, because such "indirect" memh would be always released fully by ucp_memh_put
probably could also remove map_refcount by this method